components/arm/arm.go

```diff
@@ -365,34 +364,13 @@ func Move(ctx context.Context, r robot.Robot, a Arm, dst spatialmath.Pose) error
 
 // Plan is a helper function to be called by arm implementations to abstract away the default procedure for using the
 // motion planning library with arms.
-func Plan(
-	ctx context.Context,
-	r robot.Robot,
-	a Arm,
-	dst spatialmath.Pose,
-) ([][]referenceframe.Input, error) {
-	// build the framesystem
-	fs, err := framesystem.RobotFrameSystem(ctx, r, nil)
-	if err != nil {
-		return nil, err
-	}
-	armName := a.ModelFrame().Name()
-	destination := referenceframe.NewPoseInFrame(armName+"_origin", dst)
-
-	// PlanRobotMotion needs a frame system which contains the frame being solved for
-	if fs.Frame(armName) == nil {
-		armFrame := a.ModelFrame()
-		jp, err := a.JointPositions(ctx, nil)
-		if err != nil {
-			return nil, err
-		}
-		return motionplan.PlanFrameMotion(ctx, r.Logger(), dst, armFrame, armFrame.InputFromProtobuf(jp), defaultArmPlannerOptions, nil)
-	}
-	solutionMap, err := motionplan.PlanRobotMotion(ctx, destination, a.ModelFrame(), r, fs, nil, defaultArmPlannerOptions, nil)
+func Plan(ctx context.Context, logger golog.Logger, a Arm, dst spatialmath.Pose) ([][]referenceframe.Input, error) {
+	armFrame := a.ModelFrame()
+	jp, err := a.JointPositions(ctx, nil)
 	if err != nil {
 		return nil, err
 	}
-	return motionplan.FrameStepsFromRobotPath(a.ModelFrame().Name(), solutionMap)
+	return motionplan.PlanFrameMotion(ctx, logger, dst, armFrame, armFrame.InputFromProtobuf(jp), defaultArmPlannerOptions, nil)
```

Comment:
[nit] the last parameter of `PlanFrameMotion` is `planningOpts map[string]interface{}`.
Almost always we leave this as `nil`. only in `doftbot_test.go` is it not nil. 
Can we refactor this function such that we don't have the dangling argument?
I can see how we would want to include this for easy hacking.... so I am not opposed to keeping this argument.

Reply:
This argument gets used in the Planager™️ and is the way we are able to get different behavior from the planner, so even if its not in use now I think its still useful.

Reply:
I don't believe this is a good driver for that change, but I'm marking this for later follow-up and discussion.